api/routes/Auth.js

```diff
@@ -0,0 +1,201 @@
+                    );
+                  }
+                }
+              );
+            }
+
+            // Include fields from the User model that should
+            // be passed to the JSON Web Token (JWT)
+            const userToBeSigned = {
+              firstName: user.firstName,
+              lastName: user.lastName,
+              email: user.email,
+              accessLevel: user.accessLevel,
+              pagesPrinted: user.pagesPrinted
+            };
+            const token = jwt.sign(
+              userToBeSigned, config.secretKey, jwtOptions
+            );
+            res.status(OK).send({ token: 'JWT ' + token });
+          } else {
+            // Unauthorized
+            logger.log('User/pass doesn\'t match our records: ', user);
+            res.status(UNAUTHORIZED).send({
+              message: 'Username or password does not match our records.'
+            });
+          }
+        });
+      }
+    }
+  );
+});
+
+// Edit/Update a member record
+router.post('/setEmailToVerified', (req, res) => {
+  const query = { email: req.body.email };
+
+  User.updateOne(query, { emailVerified: true }, function(error, result) {
+    if (error) {
+      const info = {
+        userEmail: req.body.email,
+        errorTime: new Date(),
+        apiEndpoint: 'user/setEmailToVerified',
+        errorDescription: error
+      };
+      addErrorLog(info);
+      res.status(BAD_REQUEST).send({ message: 'Bad Request.' });
+    }
+
+    if (result.nModified < 1) {
+      return res
+        .status(NOT_FOUND)
+        .send({ message: `${req.body.queryEmail} not found.` });
+    }
+
+    return res.status(OK).send({
+      message: `${req.body.queryEmail} was updated.`
+    });
+  });
+});
+
+// Verifies the users session if they have an active jwtToken.
+// Used on the inital load of root '/'
+// Returns the name and accesslevel of the user w/ the given access token
+router.post('/verify', function(req, res) {
+  console.log(checkIfTokenValid(req));
+  if (!checkIfTokenSent(req)) {
+    return res.sendStatus(UNAUTHORIZED);
+  }
+  // else if (!checkIfTokenValid(req)) {
+  //   return res.sendStatus(UNAUTHORIZED);
+  // }
+  // else{
+  //   return res.status(OK).send(checkIfTokenValid(req));
+  // }
+
+  const decoded = checkIfTokenValid(req);
+  if (!decoded) {
+    res.sendStatus(UNAUTHORIZED);
+  } else {
+    res.status(OK).send(decoded);
+  }
+});
+
+function checkIfPageCountResets(lastLogin) {
+  if (!lastLogin) return false;
+
+  const newDate = new Date();
+  // + 1 to account for daylight savings time
+  newDate.setDate(newDate.getDate() + 1);
+  const amountOfDaysToLastSunday = newDate.getDate() - newDate.getDay();
+  const lastSundayDate = new Date();
+  lastSundayDate.setDate(amountOfDaysToLastSunday); // last sunday
+  lastSundayDate.setHours(23, 59, 59); // 11:59:59 PM
+
+  // If the last login is before last Sunday
+  // at 1 second before midnight, return true
```

Comment:
do we need these type of comments

Reply:
he pretty much copied and pasted this from `User.js`